### DISCUSSION TITLE
16' and 4' Unison Couplers

### DISCUSSION THREAD

**Discussion by heinlyfamily2001** [GrandOrgue/GoOdf#85] [Q&A] [answered]
This might sound like a crazy question but I am experimenting with a rather small organ.  I was wondering if it is possible that when the Unison Separation Off along with a 4' or 16'  coupler would play the lowest octave (16') or highest octave (4') instead of playing nothing.   Basically breaking back to the respective octave.  How would I go about doing something like that in GoOdf?  
Thanks.

**Comment by larspalo** [comment]
@heinlyfamily2001 If I understand your question correctly you should just create (3) coupler(s) for the Manual in question like this:

1. Unison Off coupler (set to Yes)
2. 16' coupler to target same Manual and a destination offset of -12
3. 4' coupler to target same Manual and a destination offset of 12

The couplers would normally be available as drawstops separately (you display them as GUI elements on the panel) so that you'd have to activate both the unison off coupler and an octave coupler. When both unison off and an octave coupler is active you'll hear the sound of the keys from the octave coupler but not from the key you pressed. For this to work you of course need to have pipes in the ranges of the couplers for it to have any effect.

If your question was about having manual extensions that only can be played by the octave couplers and not shown as keys, then that's a bit different story to create and you of course need to supply samples for the extensions keys/pipes too.

**Reply by heinlyfamily2001** [reply]
Thanks for your quick response.  I am trying to do the latter about manual extensions.  What I did so far was copied and renamed the 036-047 files and renamed them 024-035 and also copied and renamed the 085-096 to 097-108.  When I have the Unision Off  and the 16' or 4' coupler activated, what I would like the organ to do is when I have the Unison Off  and the 16' coupler activated is play the lowest octave and when I have the Unison Off and the 4' coupler activated I would like it to play the highest octave.  I am thinking that might be called "breaking back."  I hope that makes sense.  It is a small 6 rank organ.

**Comment by larspalo** [comment]
@heinlyfamily2001 Ok, re-reading your post I realize you asked about the latter part of my earlier answer that I didn't explain further so here it comes. You don't need to manipulate the samples themselves at all.

You need to create additional keys in the manual first. Say that the displayed compass is the normal 61 keys and you want an octave below and an octave above - then the total number of keys in the manual should be 85. Since the first playable key is one octave above the lowest available one you have to set "First accessible key logical key number" to 13 and keep the number of accessible keys to 61. Adjust all values according to your actual compass both here and in the below example!

Next, when creating the stops you need to set the "Number of logical pipes" to 85 on the internal rank tab and on the stop properties tab the "Number of accessible pipes" to 85. When you load the actual pipes into the stop internal rank you'd best use the "Flexible pipe loading..." feature. This needs to be done in several steps.

Select first pipe and push "Flexible pipe loading..." button. Set total pipes to load to 12 and then push "Load pipes" button. This will load the first octave into the "sub octave" and will thus be an octave "wrong". You could either adjust the HarmonicNumber (an octave higher than the nominal, so for an 8' original pipe you should have 16 as value since it now would play as a 4') for the pipes, or if you want it to sound like the sub-octave you add PitchTuning -1200 to the pipes (in which case you leave HarmonicNumber as it is). Either way, the fastest procedure is to select first pipe and hit Ctrl+E to enter pipe properties dialog. Make adjustments to either value to match your expectations and then increase pipes number spin ctrl at the bottom to 11 and then click "Copy current properties to next" button.

Select the 13th pipe and push"Flexible pipe loading..." button. Set total pipes to load to 61 and make sure that the 13th pipe will still match MIDI number 36. Push "Load pipes" button.

Finally select the 74th pipe and push"Flexible pipe loading..." button. Set the MIDI number to match to one octave lower - would in my example mean 85 and the number of pipes should be calculated automatically (but it should be 12). Click "Load pipes" button. Now with the 74th pipe selected you must adjust either HarmonicNumber or the PitchTuning to match in reverse to the lowest octave. HN as a 16' for a normal 8' would mean a value of 4. PitchTuning to raise the pitch an octave would be 1200. Increase the pipes spin ctrl to 11 and click the "Copy current properties to next" button.

Hopefully by now you have this working... I'll try to remember to create a screencast about this as it's by no means obvious!

**Reply by heinlyfamily2001** [reply, marked as answer]
Thank you for your quick response once again.  It worked like a charm.  Your directions were very clear and easy to follow.  

I like the screencasts that you do.  I watched all of them so far and they have helped me a great deal.  They are easy to follow along with.  I like the step by step process you show.

Thanks again for your assistance in this matter.

**Comment by larspalo** [comment]
Screencast number 12 https://youtu.be/hzpuzJnttWs demonstrates this with a few little extras too - like that I discovered a small reading bug in the process of recording it... :)
